src/components/Calculator.js

```diff
@@ -1,47 +1,56 @@
 import React, { Component } from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends Component {
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    e.preventDefault();
+    this.setState((prev) => calculate(prev, e.target.name));
   }
```

Comment:
**Changes Required**
- [ ] Every time a button is clicked, the state update will re-render the `virtual DOM`. To handle this in `Class Components`, we need to usually wrap this `side-effect` inside lifecycle methods like `ComponentDidMount` and `ComponentWillUpdate`. Let us try to use lifecycle methods to handle this DOM side effect.
- [ ] The `handleClick` function should be defined in the parent component `App.js` and passed to it's child `Calculator.js` as a prop. In this way, any nested child of `App.js` will be able to access it as a prop  without redefining it.